---
title: 'scRNA-seq Analysis of 10X Homeostatic Sample isl1_sib'
author: "Nicole Tran"
date: "`r format(Sys.time(), '%d %B, %Y')`"
output:
  html_document: default
  pdf_document: default
  fig_caption: yes
  pdf_document': default
---

```{r, setup}
knitr::opts_chunk$set(message = FALSE)
knitr::opts_chunk$set(warning = FALSE)

#install.packages("Seurat")
library(Seurat)
library(plotly)
#install.packages("future")
library(future)
library(dplyr)
library(ggplot2)
library(cowplot)
library(gridExtra)
library(ggrepel)
options(future.globals.maxSize = 1000 * 1024^2)

setwd("/Volumes/easystore/SIMR_2019/pio-lab/scripts")
```

```{r, data}
#load in isl1_sib_counts_10X dir containing count matrix
#files in this folder should be in .gz format
isl1_sib_10X.data <- Read10X(data.dir = "../data/isl1_sib_counts_10X/")

homeo.isl1_sib_10X <- CreateSeuratObject(counts = isl1_sib_10X.data, project = "homeo.isl1.sib.10X", min.cells = 1, min.features = 1)

homeo.isl1_sib_10X
```
#### Standard pre-processing workflow

```{r, mt}
#calculate mitochondrial contamination
#do not add suffix to identify sample in "percent.mt" for metadata, will cause convergance issues in downstream integration with multiple datasets
homeo.isl1_sib_10X[["percent.mt"]] <- PercentageFeatureSet(homeo.isl1_sib_10X, pattern = "^mt-")

```

#### QC metrics

```nFeature_RNA``` is the number of genes detected in each cell. 

```nCount_RNA``` is the total number of molecules detected within a cell. 

Low ```nFeature_RNA``` for a cell indicates that it may be dead/dying or an empty droplet. 

High ```nCount_RNA``` and/or ```nFeature_RNA``` indicates that the "cell" may in fact be a doublet (or multiplet). 

```{r, vln, fig.height=20}
#vlnplots of number of genes, molecule count, and mitochondrial % in cell
#summary tables used to determine minimal lower threshold nCounts_RNA for subsetting

#nFeature_RNA
#will not be delimitting threshold for nFeature_RNA, we want all possible genes
nFeature.vln.homeo.isl1.sib.10X <-VlnPlot(object = homeo.isl1_sib_10X, features = "nFeature_RNA") + geom_hline(yintercept = 3000, color = 'red') + geom_hline(yintercept = 200, color = 'red') +  scale_y_continuous(breaks=c(200, 3000))
summary(homeo.isl1_sib_10X$nFeature_RNA)

#nCount_RNA
nCount.vln.homeo.isl1.sib.10X <- VlnPlot(object = homeo.isl1_sib_10X, features = "nCount_RNA") + ylim(0, 15000) + geom_hline(yintercept = 10000, color = 'red') 
nCount.vln.homeo.isl1.sib.10X
summary(homeo.isl1_sib_10X$nCount_RNA)

#percent.mito %
pct.mito.vln.homeo.isl1.sib.10X <- VlnPlot(object = homeo.isl1_sib_10X, features = "percent.mt") + geom_hline(yintercept = 10, color = 'red') + scale_y_continuous(breaks=10)
summary(homeo.isl1_sib_10X$percent.mt)

#CombinePlots(plots = list(nFeature.vln.homeo.isl1.sib.10X + nCount.vln.homeo.isl1.sib.10X + pct.mito.vln.homeo.isl1.sib.10X), ncol = 3)

nFeature.vln.homeo.isl1.sib.10X + nCount.vln.homeo.isl1.sib.10X + pct.mito.vln.homeo.isl1.sib.10X
```


```{r}
boxplot(homeo.isl1_sib_10X$nFeature_RNA, lwd= 2, ylim = c(0,500))
stripchart(homeo.isl1_sib_10X$nFeature_RNA, vertical = TRUE, 
    method = "jitter", add = TRUE, pch = 20, col = 'blue', bg= 'black')

#head(homeo.isl1_sib_10X$nCount_RNA)
```

```{r, fig.width = 10, featurescatter}
featurescatter.nCountXpercent.mt.homeo.isl1.sib.10X <- FeatureScatter(homeo.isl1_sib_10X, feature1 = "nCount_RNA", feature2 = "percent.mt") + xlim(0, 15000)
featurescatter.nCountXnFeature.homeo.isl1.sib.10X <- FeatureScatter(homeo.isl1_sib_10X, feature1 = "nCount_RNA", feature2 = "nFeature_RNA") + xlim(0, 15000)
featurescatter.nCountXpercent.mt.homeo.isl1.sib.10X + featurescatter.nCountXnFeature.homeo.isl1.sib.10X

```

#### Subsetting

```{r, subset}
#subsetting parameters: omit cells with genes less than 3000 and greater than 200. omit mitochondrial contamination greater than 10%. omit cells with molecules greater than 10,000
homeo.isl1_sib_10X <- subset(homeo.isl1_sib_10X, subset = nFeature_RNA > 200 & nFeature_RNA < 3000 & percent.mt < 10)

homeo.isl1_sib_10X
#2979 cells preserved from 17713 in original data
```

Noticable increase in cells/samples from 2979 to ~16000 using an alternative subsetting parameter:
```homeo.isl1_sib_10X <- subset(homeo.isl1_sib_10X, subset = nFeature_RNA > 200 & nFeature_RNA < 3000 & percent.mt < 10 & nCount_RNA <10000)```


```{r, normalize}
#call the plan() function to initiate the multiprocess/parallelization for NormalizeData()
plan("multiprocess")
homeo.isl1_sib_10X <- NormalizeData(homeo.isl1_sib_10X, normalization.method = "LogNormalize", scale.factor = 10000, verbose = TRUE)
#mutliprocess for NormalizationData using 1G is executable, not same case for ScaleData()
```

```{r, FindVariableFeatures}
#Identification of highly variable genes
homeo.isl1_sib_10X <- FindVariableFeatures(homeo.isl1_sib_10X, selection.method = "vst", nfeatures = 2000)

# Identify the 10 most highly variable genes
top10 <- head(VariableFeatures(homeo.isl1_sib_10X), 10)

#plot 2000 variable features with labels on top 10 most variable genes
VariableFeatureplot1.homeo.isl1_sib_10X <- VariableFeaturePlot(homeo.isl1_sib_10X) 
VariableFeature.labedlplot2.homeo.isl1_sib_10X <- LabelPoints(plot = VariableFeatureplot1.homeo.isl1_sib_10X, points = top10, repel = TRUE) + ggtitle("Top 10 Variable Genes Using VariableFeaturePlot")

VariableFeature.labedlplot2.homeo.isl1_sib_10X

```

```{r, scale}
all.genes <- rownames(homeo.isl1_sib_10X)
#25622 genes total
#another method of accessing unique, nonduplciated gene symbols in counts data.
#returns 0 if there are no duplicates
test<- all.genes %in% all.genes[duplicated(all.genes)] 
sum(grepl(pattern = "TRUE", x = test))

plan("sequential")
#options(future.globals.maxSize = 3500 * 1024^2)
#plan("multiprocess")
homeo.isl1_sib_10X <- ScaleData(homeo.isl1_sib_10X, features = all.genes, vars.to.regress = "percent.mt", verbose = TRUE)
```

Error in getGlobalsAndPackages(expr, envir = envir, globals = globals) : 
  The total size of the 11 globals that need to be exported for the future expression (‘FUN()’) is 3.18 GiB. This exceeds the maximum allowed size of 0.98 GiB (option 'future.globals.maxSize'). The three largest globals are ‘object’ (3.17 GiB of class ‘numeric’), ‘features’ (1.67 MiB of class ‘character’) and ‘split.cells’ (1.39 MiB of class ‘list’).
  
  Formerally using set future.global to 1G, change to 3.5G however activit usage on local mac is using 6G/8G. If R session aborts, will have to resort to "sequential" and fargo parallelization.
```{r, save}
#default save RData env
save.image("../data/workspace_homeo_isl1_sib_10X.RData")
#3.1G, this seems to be the correct way
```

### Perform PCA

```{r, RunPCA}
#default pc is set to 100
homeo.isl1_sib_10X <- RunPCA(homeo.isl1_sib_10X, npcs = 100, verbose = TRUE, features = all.genes)

VizDimLoadings(homeo.isl1_sib_10X, dims = 1:2, reduction = "pca")

DimHeatmap(homeo.isl1_sib_10X, dims = 1:15, cells = 500, balanced = TRUE)

DimPlot(homeo.isl1_sib_10X, reduction = "pca")

```

### Choosing PC
```{r,scree}
ElbowPlot(homeo.isl1_sib_10X, ndims = 50)

ElbowPlot(homeo.isl1_sib_10X, ndims = 30)

ElbowPlot(homeo.isl1_sib_10X, ndims = 20)

```
Scree plot shows that variance/eigenvalues in data seem to level off between 15-20 PCs. We'll get the most information within the first ~15 PCs. Use jackstraw to confirm statistical power.

```{r, jackstraw}
#dims is defaulted to 20
homeo.isl1_sib_10X <- JackStraw(homeo.isl1_sib_10X, num.replicate = 100)
homeo.isl1_sib_10X <- ScoreJackStraw(homeo.isl1_sib_10X, dims = 1:20)
JackStrawPlot.PC20.homeo.isl1_sib_10X<- JackStrawPlot(homeo.isl1_sib_10X, dims = 1:20)
```
Showing less statistical significance past >18 PC. Will set PC to 15

### Clustering/UMAP

```{r, cluster}
homeo.isl1_sib_10X <- FindNeighbors(homeo.isl1_sib_10X, dims = 1:15, features = all.genes)

homeo.isl1_sib_10X <- FindClusters(homeo.isl1_sib_10X, resolution = 1.2)

homeo.isl1_sib_10X <- RunUMAP(homeo.isl1_sib_10X, dims = 1:15, reduction = "pca")

DimPlot(homeo.isl1_sib_10X, reduction = "umap", label = TRUE) 


```

PC 15 with resolution 1.2, 14 clusters: gave pretty neat cluster, some single cells outside their prespective groupings however no striking outliers

```{r}
save.image("../data/workspace_homeo_isl1_sib_10X.RData")
```

### Annotate Clusters

```{r, gene_table}
#import gene info list pulled from ensembl
gene_table <- read.table("../data/Danio_Features_unique_Ens98_v1.tsv", sep = "\t", header = TRUE)

#change header name from 
```

```{r, findallmarkers}
#mirrored as closely in parameters to Lush, 2019. as possible (see "Quality control, dimensional reduction, and cell classification" under Methods)
#fold change greater than 0.10, or less than 􏰀0.10
#only return p-value <.10 (see return.thres())
all.markers.homeo.isl1_sib_10X <- FindAllMarkers(homeo.isl1_sib_10X, only.pos = FALSE, min.pct = 0.10, logfc.threshold = 0.10, return.thresh = 0.01, verbose = TRUE)

```
```{r}
#change the column name formerally called "gene" generated from FindAllMarkers() to  "Gene.name.uniq" so we can merge with gene_table
#must have same column name in order to match genes
colnames(all.markers.homeo.isl1_sib_10X)[7] <- "Gene.name.uniq"

#inspect new col name change
names(all.markers.homeo.isl1_sib_10X)

all.markers.homeo.isl1_sib_10X <- merge(all.markers.homeo.isl1_sib_10X, gene_table, by = "Gene.name.uniq") 


all.markers.homeo.isl1_sib_10X <- all.markers.homeo.isl1_sib_10X[order( all.markers.homeo.isl1_sib_10X[,7], all.markers.homeo.isl1_sib_10X[,3] ),]


```

```{r, allmarker}
#export results corresponding to each cluster into respective .tsv files

```
